ui.admin-mui/src/components/Settings/Import/nightbot.tsx

```diff
@@ -65,29 +66,124 @@ const PageSettingsModulesImportNightbot: React.FC<{
     enqueueSnackbar('User access revoked.', { variant: 'success' });
   }, [ enqueueSnackbar ]);
 
-  return (<Box ref={ref} id="nightbot">
-    <Typography variant='h2' sx={{ pb: 2 }}>Nightbot</Typography>
-
-    <Paper elevation={1} sx={{ p: 1 }}>
-      {userLoadInProgress}
-      <TextField
-        disabled
-        fullWidth
-        variant='filled'
-        value={userLoadInProgress ? 'Loading user data...' : user}
-        label={translate('integrations.lastfm.settings.username')}
-        InputProps={{
-          startAdornment: userLoadInProgress && <InputAdornment position="start"><CircularProgress size={20}/></InputAdornment>,
-          endAdornment:   <InputAdornment position="end">
-            { user !== 'Not Authorized'
-              ? <Button color="error" variant="contained" onClick={revoke}>Revoke</Button>
-              : <Button color="success" variant="contained" onClick={authorize}>Authorize</Button>
+  type PlaylistItemTrack = {
+    providerId: string;
+    provider: string;
+    duration: number;
+    title: string;
+    artist: string;
+    url: string;
+  };
+
+  type PlaylistItem = {
+    track: PlaylistItemTrack;
+    _id: string;
+    createdAt: string;
+    updatedAt: string;
+  };
+
+  type PlaylistPage = {
+    status: number;
+    _sort: { date: 'asc' | 'desc' };
+    _limit: number;
+    _offset: number;
+    _total: number;
+    playlist: PlaylistItem[];
+  };
+
+  const fetchPlaylist = async (acc: PlaylistItemTrack[], offset: number): Promise<T> => {
+    const url = 'https://api.nightbot.tv/1/song_requests/playlist?limit=100&offset=';
+    let maxRetries = 3;
+
+    try {
+      const response = await axios.get(url + offset, { headers: { Authorization: 'Bearer ' + accessToken } });
+      const data: PlaylistPage = response.data;
+      const tracks = data.playlist.map((e) => e.track);
+
+      if (acc.push(...tracks) >= data._total ?? 0) {
+        return acc;
+      }
+
+      return await fetchPlaylist(acc, offset + 100);
+    } catch (error: any) {
+      if (error.response.status === 429 && maxRetries !== 0) {
```

Comment:
I think the conditional branches in here should also return the accumulator. Currently i think both else branches just fall out, returning `undefined` as a default.